src_c/collisions_avx2.c

```diff
@@ -0,0 +1,90 @@
+#include "include/pygame.h"
+#include "include/geometry.h"
+
+#include <immintrin.h>
+
+
+static PG_FORCEINLINE int
+pgIntersection_LineRect_avx2(pgLineBase *line, SDL_Rect *rect, double *X, double *Y, double *T) {
```

Comment:
Could you you use better names here? line xintersect and yintersect for example? T as well